### DISCUSSION TITLE
e!cockpit problem

### DISCUSSION THREAD

**Discussion by Jarnoklein** [MichielVanwelsenaere/HomeAutomation.CoDeSys3#89] [Q&A]
hello,

I'm using your e!cockpit project, but i have a problem
Only at the startup the mqtt messages were send. Not when i change an output or something.
controlling outputs with an mqtt message is possible, but i can't get an actual state out of it.

what am i doing wrong?

kind regards,

Jarno

**Comment by MichielVanwelsenaere** [comment]
Hi Jarno,

sounds like you are having a general problem sending messages from your PLC to the broker.
Could you check the logs in the errorhistory as explained over here: [link](https://github.com/stefandreyer/CODESYS-MQTT/blob/master/doc/integration.md#debugging).

Perhaps they provide some more insights to what the issue could be.

**Reply by Jarnoklein** [reply]
![image](https://user-images.githubusercontent.com/62300191/112530273-8cb9b380-8da6-11eb-8863-a1befa6bc273.png)

is this what you mean?

**Reply by MichielVanwelsenaere** [reply]
Yes, it's looking good.
What broker are you using? This feels network/broker related.. are you receiving the heartbeat messages?

**Comment by Jarnoklein** [comment]
I use the Mosquitto broker running in home assistant.
I receive the heartbeat message only 1 time when the controller is coming online.

also all the other topics only work 1 time at the startup

**Reply by MichielVanwelsenaere** [reply]
Could you
- verify the broker logs?
- what are the settings of the mqqt task? (Should be freewheeling)
-  try another broker? (mosquitto is fine but try it as a standalone component)

**Reply by Jarnoklein** [reply]
Solved the problem! you're right, it was caused by the broker. Thanks for the support so far!!

so now i've got the application running and it's working fine,
i want to expand the program with some DMX functions to control some light fixtures but then i got the next problem, i/o errors.
I use a Wago PFC100 (750-8100), 11 4channel input modules, 10 4channel output modules and a temperature sensor module. 
For the DMX signals i use a RS232/485 module. (750-652) When i add this module in between the others, and insert it in the program and let it run there are no mqtt messages send and after 20 seconds i'm getting an whatchdog timeout error, and the plc stops it's program.

can you help me solve this problem?

**Reply by MichielVanwelsenaere** [reply]
I've got little to no experience with DMX or/and an external RS485 module but it sounds like the module is causing errors on the kbus. Is your module new or did you get it somewhere 2nd hand? I would try:
- setting up your PFC100 only with the 652 module
- check with WagoIoCheck if you can read out the modules on the device
- perhaps contact wago support for more in depth troubleshooting on this one (reverting with results/feedback from them over here is appreciated)
